waves/src/App.tsx

```diff
@@ -184,42 +174,32 @@ function App() {
         }
     }, [path, history]);
 
-    const [[transaction, trade], setTransaction] = useState<[string, Trade]>(["", {} as any]);
     const [state, dispatch] = useReducer(reducer, initialState);
 
     const rate = useSSE("rate", {
         ask: 33766.30,
         bid: 33670.10,
     });
 
-    let { isOpen: isUnlockWalletOpen, onClose: onUnlockWalletClose, onOpen: onUnlockWalletOpen } = useDisclosure();
-    let { isOpen: isCreateWalletOpen, onClose: onCreateWalletClose, onOpen: onCreateWalletOpen } = useDisclosure();
-    let { isOpen: isConfirmSwapOpen, onClose: onConfirmSwapClose, onOpen: onConfirmSwapOpen } = useDisclosure();
-    let { isOpen: isWalletOpen, onClose: onWalletClose, onOpen: onWalletOpen } = useDisclosure();
-
-    let { data: getWalletStatusResponse, isLoading, reload: reloadWalletStatus } = useAsync({
+    let { data: getWalletStatusResponse, reload: reloadWalletStatus } = useAsync({
         promiseFn: getWalletStatus,
     });
-    let walletStatus = getWalletStatusResponse || { exists: false, loaded: false };
 
-    let { data: getBalancesResponse, mutate: reloadWalletBalances } = useSWR(
-        () => walletStatus.loaded ? "wallet-balances" : null,
-        () => getBalances(),
-        {
-            refreshInterval: 5000,
-        },
-    );
-    let balances = getBalancesResponse || [];
+    useEffect(() => {
+        let callback = (_message: MessageEvent) => {};
+        // @ts-ignore
+        if (!window.wallet_status) {
+            callback = async (message: MessageEvent) => {
+                debug("Received message: %s", message.data);
+                await reloadWalletStatus();
+            };
+        }
+        window.addEventListener("message", callback);
 
-    let btcBalanceEntry = balances.find(
-        balance => balance.ticker === Asset.LBTC,
-    );
-    let usdtBalanceEntry = balances.find(
-        balance => balance.ticker === Asset.USDT,
-    );
+        return () => window.removeEventListener("message", callback);
+    });
```

Comment:
This does not seem to work as expected, i.e. the website is not refreshing once the wallet was unlocked.